src/Microsoft.AspNet.Mvc.Core/Rendering/Html/HtmlHelper.cs

```diff
@@ -351,6 +351,35 @@ public string DisplayText(string expression)
         }
 
         /// <inheritdoc />
+        public IEnumerable<SelectListItem> GetSelectList<TEnum>() where TEnum : struct
+        {
+            var type = typeof(TEnum);
+            var metadata = MetadataProvider.GetMetadataForType(type);
+            if (!metadata.IsEnum || metadata.IsFlagsEnum)
+            {
+                throw new ArgumentException(
+                    Resources.FormatHtmlHelper_TypeNotSupported(type.FullName),
+                    nameof(TEnum));
```

Comment:
the `<exception/>` doc comments aren't good enough for you?  :smile_cat: 

how about I add `Type must be an enum that does not have an associated FlagsAttribute.` to the resource?  I'd pass in `nameof(Enum).ToLowerInvariant()` and `nameof(FlagsAttribute)`